src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -315,9 +324,26 @@ MultiLogicalPlanOptimize(MultiTreeRoot *multiLogicalPlan)
 	ListCell *collectNodeCell = NULL;
 	ListCell *tableNodeCell = NULL;
 	MultiNode *logicalPlanNode = (MultiNode *) multiLogicalPlan;
+	bool requiresIntermediateRowPullUp = RequiresIntermediateRowPullUp(logicalPlanNode);
```

Comment:
Other, similar properties (e.g. pullDistinctColumns) are determined in BuildExtendedOpNodeProperties. Should we call this from there for consistency?

Reply:
No, since this call alone doesn't determine the value

Reply:
Can we move all the logic that determines the value into BuildExtendedOpNodeProperties?

Reply:
Yes, but that's as far reaching as `GetAggregateType`